lib/matplotlib/mlab.py

```diff
@@ -210,6 +211,30 @@ def detrend_linear(y):
     return y - (b*x + a)
 
 
+def _stride_windows(x, n, noverlap=0):
+    if noverlap >= n:
+        raise ValueError('noverlap must be less than n')
+    if n < 1:
+        raise ValueError('n cannot be less than 1')
+
+    x = np.asarray(x)
+
+    if n == 1 and noverlap == 0:
```

Comment:
I think the special-case here can go away, it should be handled by the general case below.